### ISSUE TITLE
Merging GeoTIFFs update into an existing file

### ISSUE THREAD

**Issue by laxnpander** [laxnpander/OpenREALM#24]
Currently the global map is either saved as a whole at the end of the mapping process, or for every iteration a separate geotiff is created. This does not really follow the real-time criteria, because with bigger map the saving time will increase. It could be improved, when the current map update would be merged into the existing global map. 

Steps:
1) Timing of the current saving process to get a reference
2) Implement merging
3) Timing of the merging process

**Comment by laxnpander** [comment]
1. Saving process for full save of the map on HDD (not SSD): ~0.83s per image. If each channel is saved independently, then this makes 2.4s just for the ortho photo! Currently it is running in the mosaicing thread, so it will block the mosaicing as well.

[stage.log](https://github.com/laxnpander/OpenREALM/files/4836275/stage.log)

-> Very slow actually. This is not viable for future use. Either we need a driver which accepts saving requests only once the last one is processed in a separate thread, or the saving process must be optimized. Or maybe even both. Let's see how fast the incremental update is first.

**Comment by laxnpander** [comment]
2. Unfortunately as it turns out, the merging process provided by GDAL seems to be limited to creating a new file. Therefore the merge first reads the existing file, takes the additional data and writes both of them in a new file. If applied on the original file, the data is added but the image boundaries are not extended. 

Conclusion: There is likely no existing way of dynamically extending one .tif.

**Comment by laxnpander** [comment]
Implemented a GTiff writing module, that runs in a separate thread and polls save requests. Should be sufficient for now.
